### PR TITLE
[CIT-521] Add checks if plugins are executables

### PR DESCRIPTION
**Pull request by makr11st**
## Proposed changes

Check if plugin is executable and can perform list operation before adding it to the list of valid plugins.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: sm/plugin_sm/src/plugin_manager.rs [hunk 1]
```diff
@@ -68,9 +76,34 @@ impl ExternalPlugins {
...
+            dbg!(&path);
             if path.is_file() {
-                // TODO check the file is exec
-                // TODO check the command is actually a plugin
+                match Command::new(&path).arg("list").status() {
+                    Ok(code) if code.success() => {}
+
+                    // If the file is not executable or returned non 0 status code we assume it is not a valid and skip further processing.
+                    Ok(_) => {
+                        info!("File {} in plugin directory does not support list operation and may not be a valid plugin, skipping.", pathbuf_to_string(path.clone()).unwrap());
```
Comment (rina23q) [inline sm/plugin_sm/src/plugin_manager.rs:86]: It's actually an error, so I suggest to use `error!` instead of `info!`. Also below.

**Review by rina23q — APPROVED** [review]
Just a minor comment.

File: sm/plugin_sm/src/plugin_manager.rs [hunk 2]
```diff
@@ -68,9 +76,34 @@ impl ExternalPlugins {
...
+                        continue;
+                    }
+
+                    Err(err) if err.kind() == ErrorKind::PermissionDenied => {
+                        info!(
+                            "File {} Permission Denied, is the file executable?",
+                            pathbuf_to_string(path.clone()).unwrap()
+                        );
```
Comment (didier-wenzek) [inline sm/plugin_sm/src/plugin_manager.rs:94]: I suggest to use `error!` and to explicitly tell that the file is not used as a plugin.

File: sm/plugin_sm/src/plugin_manager.rs [hunk 3]
```diff
@@ -68,9 +76,34 @@ impl ExternalPlugins {
...
+                        continue;
+                    }
+
+                    Err(err) => {
+                        info!(
+                            "An error occurred while trying to run: {}: {}",
+                            pathbuf_to_string(path.clone()).unwrap(),
+                            err
+                        );
```
Comment (didier-wenzek) [inline sm/plugin_sm/src/plugin_manager.rs:103]: I suggest to use error! and to explicitly tell that the file is not used as a plugin.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**